

## Description
This folder should be used as working directory for PhaME.

## Content

- **README_phame.md** : *This file*

## Contet to be added

- **Results:** *(folder)* Will be generated by PhaME to store Results
- **contigs:** *(files)* The resulting contic files form SPAdes should be placed here . 
- **files:** *(folder)* Will be generated by PhaME to hold the used files. 

## Running Phame
The second step was to run PhaME [[1]](#1) which provided us with a phylogenetic tree of all isolates. The same output can be generated by downloading and using PhaME on the previously described contig files. For this please visit the [PhaME Git hub repository](https://github.com/LANL-Bioinformatics/PhaME) and follow their instructions. In the folder *code* our control file can be found (221111_phame_all_final_RaxML.ctl). We used *221111_phame_all_final_RaxML* as our working directory. Doing the same will allow for the R-scripts to properly source the correct files down the line. We used PhaME version number **xxx**



## Bibliography

<a id="1">[1]</a>
Shakya M, Ahmed SA, Davenport KW, Flynn MC, Lo C-C, Chain PSG. Standardized phylogenetic and molecular evolutionary analysis applied to species across the microbial tree of life. *Sci Rep* 2020; **10**: 1723


